libs/sdk-ui-geo/src/core/geoChart/helpers/geoChart/data.ts

```diff
@@ -55,15 +63,45 @@ export function getLocation(latlng: string | null): IGeoLngLat | null {
     };
 }
 
+export function parseCoordinate(coordinate: string | null): number | null {
+    if (!coordinate) {
+        return null;
+    }
+
+    const numericalCoordinate = dataValueAsFloat(coordinate);
+    if (isNaN(numericalCoordinate)) {
+        // eslint-disable-next-line no-console
+        console.warn(
+            "UI-SDK: geoChartDataSource - getLocation: invalid latitude/longitude coordinate",
+            numericalCoordinate,
+        );
+        return null;
+    }
+
+    return numericalCoordinate;
+}
+
+interface IGeoDataBuckets {
```

Comment:
New type used only in the data parsing function to transport parsed latitude and longitude from `getBucketItemNameAndDataIndex` function. Data will not be necessary in inner parts of the chart where already parsed latitude and longitude is used (see the comment below) therefore `IGeoData` interface was not "polluted" with these props.